### PR TITLE
Build image with systemtap and profiling tools

### PR DESCRIPTION
**Pull request by p0pr0ck5**
Something we'd be interested in?

### CODE REVIEW

**Comment by Tieske** [comment]
Would need some description on how to use it in the readme

another question I have with this image is whether every thing should be installed by default. Already it pulls in java and cassandra, which takes a long time. Then again, those are needed to run the testsuite, hence I added them instead of providing a script to add them later.

But probably as the concept here is simple-all-in-one, we should stick to including all at the cost of some extra time while provisioning .

**Comment by p0pr0ck5** [comment]
I'm not terribly familiar with vagrant; could we define a CLI flag or some such to build these conditionally? So it's not built by default if people don't want/need it? I'll poke the vagrant docs, but if you have experience with something like that, that'd be great :)

File: README.md [hunk 1]
```diff
@@ -221,6 +227,7 @@ environment variables:
 | `KONG_CASSANDRA`| the major Cassandra version to use, either `2` or `3`                     | `3`, or `2` for Kong versions `9.x` and older |
 | `KONG_PATH`     | the path to mount your local Kong source under the guest's `/kong` folder | `./kong`, `../kong`, or nothing. In this order. |
 | `KONG_PLUGIN_PATH` | the path to mount your local plugin source under the guest's `/kong-plugin` folder | `./kong-plugin`, `../kong-plugin`, or nothing. In this order. |
+| KONG_PROFILING  | boolean determining whether or not to build systemtap and friends tools   | undefined |
```
Comment (Tieske) [inline README.md]: formatting, the name should be quoted

File: provision.sh [hunk 2]
```diff
@@ -81,8 +82,32 @@ sudo apt-get install -y netcat openssl libpcre3 dnsmasq procps perl
...
 
+
+###########################
+# Install profiling tools #
+###########################
+if [ -n "$KONG_PROFILING" ]; then
+  # install systemtap
+  # https://openresty.org/en/build-systemtap.html
+  sudo apt-get install -y build-essential zlib1g-dev elfutils libdw-dev gettext
+  wget http://sourceware.org/systemtap/ftp/releases/systemtap-3.0.tar.gz
```
Comment (Tieske) [inline provision.sh]: it now dumps the binary output in the terminal, so add a silent switch?

**Comment by p0pr0ck5** [comment]
@Tieske ping :)

**Pull request merged**